lib/postgres_ext/active_record/relation/query_methods.rb

```diff
@@ -1,70 +1,113 @@
 module ActiveRecord
   module QueryMethods
     class WhereChain
-      def overlap(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].overlap(value))
-        end
-        @scope
+      def overlap(opts, *rest)
+        substitute_comparisons(opts, rest, Arel::Nodes::Overlap, 'overlap')
       end
 
-      def contained_within(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].contained_within(value))
-        end
-
-        @scope
+      def contained_within(opts, *rest)
+        substitute_comparisons(opts, rest, Arel::Nodes::ContainedWithin, 'contained_within')
       end
 
-      def contained_within_or_equals(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].contained_within_or_equals(value))
-        end
-
-        @scope
+      def contained_within_or_equals(opts, *rest)
+        substitute_comparisons(opts, rest, Arel::Nodes::ContainedWithinEquals, 'contained_within_or_equals')
       end
 
-      def contains(opts)
-        opts.each do |key, value|
-          @scope = @scope.where(arel_table[key].contains(value))
+      def contains(opts, *rest)
```

Comment:
Could you break this down into separate methods to make it a bit easier to follow? Specifically lines 20-44. Descriptive method names will make this easier to grok and maintain in the future. Freaking Arel :rage: :stuck_out_tongue: